File: samples/Lilly/tutorials/broadcaster/flow/inflow/bcast-inflow.js
jdata{
	struct currentTime{
		int year;
		int month;
		int date;
		int hour;
		int minute;
		int second;
	} myClock as broadcaster;

	inF as inflow of app://time.outF;
}

/* FLOW ACTIONS */

inF.setTerminalFunction(function(entry){
	var data = entry.data;
	console.log("Received data ......");
	prettyPrint(data);

	myClock.broadcast(data);
});

function prettyPrint(time){
	var output = String(time.year+"-"+time.month+"-"+time.date+" "+time.hour+":"+time.minute+":"+time.second);
	console.log(output);
	return output;
}

File: samples/Lilly/tutorials/broadcaster/flow/inflow/bcast-inflow.c
#include <stdio.h>
#include <unistd.h>

int main(int argc, char *argv[]){
	struct currentTime local;

	while(1){
		local = timeKeeping;
		printf("%d-%d-%d %d:%d:%d\n", local.year, local.month, local.date, local.hour, local.minute, local.second);
		sleep(1);
	}

	return 0;
}